ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/thread/ThreadActionData.java

```diff
@@ -242,7 +242,7 @@ public static void purgeActions (Thread t, boolean unregister)
 
 					if (act != null)
                     {
-						act.removeChildThread(ThreadUtil.getThreadId(t));
+						act.removeChildThread(ThreadUtil.removeThreadId(t));
                     }
```

Comment:
Same comment as below about the possibility that ThreadUtil.removeThreadId(t) can return a null which will cause removeChildThread(null) to throw an NPE. The old code guaranteed that ThreadUtil.getThreadId(t) would always return a non null value. So I think you need an if null here:

```java
String tid = ThreadUtil.removeThreadId(t)

if (tid != null)
  act.removeChildThread(tid);
```